std/experimental/color/conv.d

```diff
@@ -0,0 +1,523 @@
+// Written in the D programming language.
+
+/**
+    This module implements various _color type conversions.
+
+    Authors:    Manu Evans
+    Copyright:  Copyright (c) 2015, Manu Evans.
+    License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
+    Source:     $(PHOBOSSRC std/experimental/color/conv.d)
+*/
+module std.experimental.color.conv;
+
+import std.experimental.color;
+import std.experimental.color.rgb;
+import std.experimental.color.xyz;
+
+import std.traits: isNumeric, isIntegral, isFloatingPoint, isSigned, TemplateOf;
+import std.typetuple: TypeTuple;
+
+
+/**
+Convert between color types.
+*/
+To convertColor(To, From)(From color) if(isColor!To && isColor!From)
+{
+    // no conversion is necessary
+    static if(is(To == From))
+        return color;
+
+    // *** XYZ is the root type ***
+    else static if(isXYZ!From && isXYZ!To)
```

Comment:
> > It's still an abomination.

> no it is common practice with template constraints

But it doesn't work in this case. Template constraints don't have a fall-back feature.

> > I don't want to do that to the symbol table.

> that is no argument. are you written a linker or a maintainable, extensible library.

The library has an equal or greater responsibility not to abuse the language.

> > It wasn't an accident, it was entirely deliberate.
I couldn't work out any way to make a fall-back constraint.

> which one is it.

Both, these aren't mutually exclusive in any way.

> [...]
> that would make the template constraints really easy to read and you would have nice contained functions

I'll have a go at something like that, but I still can't see how the else block is supposed to work.
These little scope blocks are nice and contained, and headed with the same if that would otherwise just be a constraint on the end of the line anyway.
You argue like this is somehow impossible to read and outright unmaintainable. It's not that bad at all, it just looks a bit awkward, but it's plain as day where the what-would-otherwise-be-small-functions start and end. What *is* unreadable and unmaintainable, is the else block under a constraint based approach (that was my first iteration).
I just don't buy the argument that splitting it into separate functions would make the scoped blocks of code headed with clear if's any easier to read. If anything, the constraint that would be on the end of the line is even clearer and more central in this configuration.
Add consideration that the symbol table suffers far less abuse under this configuration, I'm very happy with this as is. If there's no advantage other than preference, I'm not really strongly motivated to change it, when the change results in a proven difficult to understand and maintain 'else' constraint.